### PR TITLE
Fixing a bug with http response parsing

### PR DESCRIPTION
**Pull request by kofoworola**

### CODE REVIEW

File: pkg/engine/actions/executables/http/http.go [hunk 1]
```diff
@@ -40,6 +40,24 @@ type Config struct {
 	FailIfResponseEmpty  bool              `json:"failIfResponseEmpty" yaml:"failIfResponseEmpty"`
 }
 
+// transformBody converts a json.RawMessage body into the appropriate byte slice for HTTP requests.
+// It handles three cases:
```
Comment (ghost) [inline pkg/engine/actions/executables/http/http.go:44]: The `transformBody` function robustly handles json.RawMessage for HTTP request bodies, supporting normal, double-encoded, and object/array JSON. This addresses edge cases and improves interoperability with various payload sources. Consider adding a comment or test for deeply nested double-encoded objects, if that is a possibility in your usage.

File: pkg/engine/actions/executables/http/http.go [hunk 2]
```diff
@@ -113,7 +131,7 @@ func (h *Http) Execute(ctx context.Context, filledInConfig string) (interface{},
 	if cfg.ResponsePath == "" {
```
Comment (ghost) [inline pkg/engine/actions/executables/http/http.go:131]: The fallback to returning a string if unmarshalling fails increases resilience. Consider logging the unmarshal error for traceability, as silent errors may slow debugging complex integrations.

File: pkg/engine/actions/executables/http/http_test.go [hunk 3]
```diff
@@ -93,6 +94,58 @@ func TestHttp_Execute(t *testing.T) {
 				return srv.URL
 			},
 		},
+		{
```
Comment (ghost) [inline pkg/engine/actions/executables/http/http_test.go:97]: Great addition of tests for double-encoded JSON string bodies and for preserving raw string bodies. This increases confidence that edge-case payloads are handled as expected.

File: pkg/engine/requestctx/templates.go [hunk 4]
```diff
@@ -49,8 +49,12 @@ func getFuncMap(funcMap template.FuncMap) template.FuncMap {
 // stringEscape escapes a string to be safely used in various contexts
 // by escaping special characters like quotes, backslashes, and control characters.
 // Uses Go's native strconv.Quote and removes the surrounding quotes.
-func stringEscape(s string) string {
-	quoted := strconv.Quote(s)
+func stringEscape(s interface{}) string {
```
Comment (ghost) [inline pkg/engine/requestctx/templates.go:52]: You changed `stringEscape` to accept `interface{}` and safe-cast to string. Consider returning an error or at least logging a warning when the cast fails, to ease debugging when unexpected types are passed.

**Comment by ghost** [comment]
## Code Review Overview

### Summary

This PR addresses a bug with HTTP response and request body parsing in the HTTP executable component. It introduces a robust `transformBody` function to correctly handle various forms of JSON payloads when preparing HTTP request bodies, and adapts testing to verify correctness for more complex payload edge cases. Additionally, it updates the `stringEscape` function in the template context to safely receive and escape string inputs of type `interface{}`.

---

### Enhancements

- **Comprehensive Request Body Handling:**  
  The new `transformBody` function in `http.go` elegantly handles plain strings, double-encoded JSON, and standard JSON objects/arrays. This leads to more predictable and flexible HTTP request formation.
  
- **Expanded Test Coverage:**  
  The tests in `http_test.go` now cover edge cases involving double-encoded JSON and explicitly verify correct body transformation for various string payloads, increasing confidence and regression protection.

- **Defensive Template Escaping:**  
  By updating `stringEscape` to accept an `interface{}` and only escape when the value is a string, unexpected inputs are less likely to cause runtime panics.

---

### Potential Issues

- **Silent Fallback on Unmarshal Error (`http.go`, line 131):**  
  If JSON unmarshalling fails (for response parsing), the code simply returns the raw string, potentially obscuring issues with unexpected or malformed JSON. Logging the error or surfacing the failure more explicitly could ease debugging.

- **Empty String Masking in `stringEscape` (`templates.go`, line 52):**  
  When `stringEscape` receives a non-string, it quietly returns an empty string. While defensive, this may suppress subtle bugs by not alerting the developer when the wrong type is passed. Logging or error reporting would improve traceability.

- **Edge Case Clarification for Deeply Nested or Multiple-encoded Objects (`http.go`, around line 44):**  
  While current test cases are strong, edge cases involving deeply nested or repeatedly encoded payloads aren't explicitly documented or tested. Clarifying intended support (or lack thereof) with a comment would be helpful.

---

### Other Observations

- The adoption of both new functionality and corresponding test cases reflects good engineering hygiene.
- The method comments for `transformBody` are clear and enable future maintainers to easily understand the intention and logic.
- Formatting and naming remain consistent with project conventions.

---

#### Individual file comments for actionable improvements have been added. 

**Overall:**  
Nice job addressing the edge cases with payload parsing and response handling! A few minor adjustments for error handling and documentation would further strengthen this contribution.
